internal/admission/eventsource/handler.go

```diff
@@ -62,12 +64,16 @@ func (h *Handler) Handle(ctx context.Context, req admission.Request) admission.R
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	_, ok := out.Annotations[h.annotationKey]
+	sourceValue, ok := out.Annotations[h.annotationKey]
 	if !ok {
 		log.V(1).Info("Annotation not found, ignoring eventsource")
 		return admission.Allowed("No modifications needed")
 	}
 
+	if !h.knownSources[sourceValue] {
```

Comment:
nit: existence of they key feels safer

```suggestion
	if _, ok := h.knownSources[sourceValue]; !ok {
```